Amazon S3 Bucket for upload and download images.

dellmdq opened the pull request:
Created S3Service and S3Controllers to handle image upload and download to and from S3 bucket.
Exception handling not implemented yet.


Tickets:
https://alkemy-labs.atlassian.net/browse/OT109-38
https://alkemy-labs.atlassian.net/browse/OT109-39

dellmdq committed:
Saving just in case

dellmdq committed:
FILE_NAME npe

dellmdq committed:
FILE_NAME npe

dellmdq committed:
S3 Post and Get Request Working

dellmdq committed:
probando git

dellmdq committed:
checking github config

dellmdq committed:
Git OK

dellmdq committed:
Just saving

dellmdq committed:
Save Fix. Now returns file path as a String

dellmdq committed:
Security Fix

nicotruk committed:
application.properties recovered

nicotruk commented:
El POST tendrias que modificarlo para que desde swagger se puede subir un archivo.
Esta es una captura de como se ve desde swagger tu codigo:
![Screenshot from 2021-12-02 18-42-41](https://user-images.githubusercontent.com/17879377/144509544-1a9dace6-b61f-42ef-b135-1bd529a21912.png)

Asi es como se deberia ver:
![Screenshot from 2021-12-02 18-42-03](https://user-images.githubusercontent.com/17879377/144509566-c1beeb70-9c6f-4c93-9acd-64c4d31263d0.png)


Por otro lado, la url que devuelve puesta en un browser no me devuelve la imagen. A vos te anduvo esto? El  GET tambien hay que modificarlo.

nicotruk committed:
pom.xml recovered

dellmdq commented:
Ya me pongo a ver como hacerlo funcionar desde swagger. 
Desde postman funciona todo.
La url puesta en el browser falla por los permisos. Desde postman si agregas las credenciales para acceder al servidor funcionan.

dellmdq committed:
S3Controller Swagger fix

dellmdq committed:
Pulling remote changes of amazon branch
Merge branch 'amazon' of https://github.com/alkemyTech/OT109-Server into amazon

dellmdq commented:
Swagger Fix Ready.


![S3 Swagger Demo](https://user-images.githubusercontent.com/45313969/144654400-9aa5f617-186f-4c60-b051-89ea3a382696.gif)

dellmdq committed:
Deleted Comments
